src/mesh/analysis.rs

```diff
@@ -196,28 +197,57 @@ pub fn border_vertex_indices(edge_sharing: &EdgeSharingMap) -> HashSet<u32> {
 /// Finds continuous loops of border edges, starting from a random edge.
 ///
 /// The mesh may contain holes or islands, therefore it may have an unknown
-/// number of edge loops. If two edge loops meet at a single vertex, the result
-/// may be unpredictable and erratic.
-pub fn border_edge_loops(edge_sharing: &EdgeSharingMap) -> Vec<Vec<UnorientedEdge>> {
-    let mut border_edges: Vec<_> = border_edges(edge_sharing).map(UnorientedEdge).collect();
-
-    let mut edge_loops: Vec<Vec<UnorientedEdge>> = Vec::new();
+/// number of edge loops. If only two edge loops meet at a single vertex
+/// (e.g. if the mesh is non-manifold), this border edge chain is not included
```

Comment:
Two edge loops meeting at a single point don't make the mesh non manifold! Imagine removing (from an otherwise valid, manifold, orientable, watertight mesh) two faces sharing just a single vertex.

Reply:
Ok, I'll remove that part of the comment.

Reply:
Hmm, after drawing it out, I believe the case you describe will produce a single border edge loop from both of the removed faces. Should we stabilize (in tests) and document that behavour?

Reply:
is this what you call a "chain"? because if so, we need to consider the border edge loops as two - that's what I mean by "treating such cases" in my previous comment. I know it's difficult, but apparently such cases are not rare.

Reply:
> is this what you call a "chain"? 

Not sure what you refer to. I call a "chain" a collection of edges that can be chained, but do not form a closed loop.

Ok, so what is the expected behavior? Also, since the original task was about fixing performance, maybe let's define the behavior in trello and create another task for it?

Reply:
yes, let's make a fix task.

to answer: in my described scenario, two correct edge loops should be detected. honestly, not sure how.

Reply:
We can jump on call sometimes to discuss.

Reply:
okay, ping me when you are online. let's talk about this before closing this PR